src/server.c

```diff
@@ -2825,6 +2857,109 @@ serverDb *createDatabaseIfNeeded(int id) {
     return server.db[id];
 }
 
+void removeNodeFromList(dataType previous_type, robj *keyobj) {
+    list *process_list = NULL;
+    if (previous_type == STRING_TYPE) {
+        process_list = server.string_bigkey_info;
+    } else if (previous_type == LIST_TYPE) {
+        process_list = server.list_bigkey_info;
+    } else if (previous_type == HASH_TYPE) {
+        process_list = server.hash_bigkey_info;
+    } else if (previous_type == SET_TYPE) {
+        process_list = server.set_bigkey_info;
+    } else if (previous_type == SORTED_SET_TYPE) {
+        process_list = server.zset_bigkey_info;
+    } else {
+        // TO DO later
+    }
+
+    listIter li;
+    listNode *ln;
+    listRewind(process_list, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        bigkeyEntry *node = ln->value;
+        if (sdscmp(node->key->ptr, keyobj->ptr) != 0) continue;
+        decrRefCount(node->key);
+        listDelNode(process_list, ln);
+        zfree(node);
+    }
```

Comment:
this seems somewhat inefficient. Maybe we should try and include a b+ tree structure that will maintain the order and the ability to add/remove the object.
Another thing is the rename command. is it possible we might be renaming a key and it's metadata is not updated?